packages/language-server/src/plugins/typescript/TypeScriptPlugin.ts

```diff
@@ -485,6 +337,37 @@ export class TypeScriptPlugin
     }
 
     private getLSAndTSDoc(document: Document) {
+        return this.lsAndTsDocResolver.getLSAndTSDoc(document);
+    }
+
+    private featureEnabled(feature: keyof LSTypescriptConfig) {
+        return (
+            this.configManager.enabled('typescript.enable') &&
+            this.configManager.enabled(`typescript.${feature}.enable`)
+        );
+    }
+}
+
+export class LSAndTSDocResovler {
```

Comment:
Could you move this class to its own file? This would resolve the circular dependency between `TypescriptPlugin` and `CompletionProvider`.